packages/nextjs/hooks/scaffold-eth/useDeployedContractInfo.ts

```diff
@@ -1,19 +1,30 @@
 import { useEffect, useState } from "react";
-import { useTargetNetwork } from "./useTargetNetwork";
 import { useIsMounted } from "usehooks-ts";
 import { usePublicClient } from "wagmi";
-import { Contract, ContractCodeStatus, ContractName, contracts } from "~~/utils/scaffold-eth/contract";
+import { useAllowedNetwork } from "~~/hooks/scaffold-eth";
+import { AllowedChainIds } from "~~/utils/scaffold-eth";
+import {
+  Contract,
+  ContractCodeStatus,
+  ContractName,
+  UseDeployedContractConfig,
+  contracts,
+} from "~~/utils/scaffold-eth/contract";
 
 /**
  * Gets the matching contract info for the provided contract name from the contracts present in deployedContracts.ts
  * and externalContracts.ts corresponding to targetNetworks configured in scaffold.config.ts
  */
-export const useDeployedContractInfo = <TContractName extends ContractName>(contractName: TContractName) => {
+export const useDeployedContractInfo = <TContractName extends ContractName>({
+  contractName,
+  chainId,
+}: UseDeployedContractConfig<TContractName>) => {
   const isMounted = useIsMounted();
-  const { targetNetwork } = useTargetNetwork();
-  const deployedContract = contracts?.[targetNetwork.id]?.[contractName as ContractName] as Contract<TContractName>;
+
+  const selectedNetwork = useAllowedNetwork(chainId as AllowedChainIds);
```

Comment:
Is ` as AllowedChainIds` needed here? Looks like it's not. Change it here please and also at all the similar places

Reply:
Funnily enough, `useDeployedContractInfo` does not need it, but all of the other ones do need it.

This is because in the other ones, `selectedNetwork.id` is of type `number` and is not of type `AllowedChainIds`. Thus it needs to be casted.

Reply:
<img width="1221" alt="Screenshot 2024-12-10 at 3 51 13 PM" src="https://github.com/user-attachments/assets/44011cd4-f08a-41b3-a329-b60637feafd7">

This is the error I get when I remove the castings.

Trying to work out a solution but haven't found anything yet.

Should we change the Network object's chain to an `AllowedChain`?
Any ideas?

Reply:
[b39b93f](https://github.com/scaffold-eth/scaffold-eth-2/pull/931/commits/b39b93f36b774ff921e77328d8e20f09d28a1762) that's what I meant